src/components/Topic.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { getArticlesByQuery } from "../api";
+import Card from "./Card";
+
+const Topic = () => {
+  const [articlesByTopic, setArticlesByTopic] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const topic = searchParams.get("topic");
+
+  useEffect(() => {
+    getArticlesByQuery(topic).then((result) => {
+      setArticlesByTopic(result.articles);
+    });
+  }, [searchParams]);
+
+  return (
+    <>
+      <h1>Topic</h1>
+      {articlesByTopic.map((article) => {
+        return (
+          <Card
+            key={article.article_id}
+            article_id={article.article_id}
+            topic={article.topic}
+            title={article.title}
+            article_img_url={article.article_img_url}
+            author={article.author}
+            comment_count={article.comment_count}
+            votes={article.votes}
+            created_at={article.created_at}
+          />
+        );
+      })}
+    </>
+  );
+};
```

Comment:
This component is quite similar to your `News` component, but with a different API call to get the articles and the use of `searchParams`. If you made the API call more generic so it optionally takes in a topic component, which may or may not be defined in the `searchParams`, then you could get away with just one component rather than needing `News` and `Topic`
This isn't a mandatory change, but something to consider for any refactors you may do in the future